cmd/gctsCreateRepository_test.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+func TestGctsCreateRepositorySuccess(t *testing.T) {
+
+	config := gctsCreateRepositoryOptions{
+		Host:                "http://testHost.wdf.sap.corp:50000",
+		Client:              "000",
+		Repository:          "testRepo",
+		Username:            "testUser",
+		Password:            "testPassword",
+		RemoteRepositoryURL: "https://github.com/org/testRepo",
+		Role:                "SOURCE",
+		VSID:                "TST",
+	}
+
+	t.Run("creating repository locally successfull", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: `{
+			"repository": {
+				"rid": "com.sap.cts.example",
+				"name": "Example repository",
+				"role": "SOURCE",
+				"type": "GIT",
+				"vsid": "GI7",
+				"status": "READY",
+				"branch": "master",
+				"url": "https://github.com/git/git",
+				"version": "1.0.1",
+				"objects": 1337,
+				"currentCommit": "f1cdb6a032c1d8187c0990b51e94e8d8bb9898b2",
+				"connection": "ssl",
+				"config": [
+					{
+						"key": "CLIENT_VCS_URI",
+						"value": "git@github.wdf.sap.corp/example.git"
+					}
+				]
+			},
+			"log": [
+				{
+					"time": 20180606130524,
+					"user": "JENKINS",
+					"section": "REPOSITORY_FACTORY",
+					"action": "CREATE_REPOSITORY",
+					"severity": "INFO",
+					"message": "Start action CREATE_REPOSITORY review",
+					"code": "GCTS.API.410"
+				}
+			]
+		}`}
+
+		err := createRepository(&config, nil, nil, &httpClient)
+
+		if assert.NoError(t, err) {
+
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "http://testHost.wdf.sap.corp:50000/sap/bc/cts_abapvcs/repository?sap-client=000", httpClient.URL)
+			})
+
+			t.Run("check method", func(t *testing.T) {
+				assert.Equal(t, "POST", httpClient.Method)
+			})
+
+			t.Run("check user", func(t *testing.T) {
+				assert.Equal(t, "testUser", httpClient.Options.Username)
+			})
+
+			t.Run("check password", func(t *testing.T) {
+				assert.Equal(t, "testPassword", httpClient.Options.Password)
+			})
+
+		}
+
+	})
+
+	t.Run("repository already exists locally", func(t *testing.T) {
```

Comment:
"locally" is confusing since it mean "on the ABAP server". Has already been adjusted in the step, should be aligned here.